tool/src/access/lpc/linux.rs

```diff
@@ -0,0 +1,183 @@
+use std::{
+    fs,
+    io::{
+        self,
+        Read,
+        Write,
+        Seek,
+        SeekFrom,
+    },
+    os::unix::io::AsRawFd,
+    path::Path,
+    time::Duration,
+};
+
+use crate::{
+    Access,
+    Error,
+    StdTimeout,
+    Timeout,
+    timeout,
+};
+
+use super::*;
+
+struct PortLock {
+    start: u16,
+    len: u16,
+    file: fs::File,
+}
+
+impl PortLock {
+    pub fn new(start: u16, end: u16) -> io::Result<Self> {
```

Comment:
Maybe this should be `start` and `len`, since the caller has a start and a length, and the code here has to convert it to a length anyway.

Not hugely important though.

Reply:
The reason I chose end is to avoid potential overflows